backend/app/users/feedback/service.py

```diff
@@ -0,0 +1,105 @@
+
+from app.users.feedback.model import FeedbackRecordModel
+from app.users.auth import UserInfo
+from app.server_dependencies.database_collections import Collections
+from app.users.repositories import UserPreferenceRepository
+from app.constants.errors import ErrorService
+
+# Cache for questions data
+questions_cache: Dict[str, Any] = {}
+
+
+async def load_questions() -> Dict[str, Any]:
+    global questions_cache
+    if not questions_cache:
+        questions_file = Path(__file__).parent / "questions-en.json"
+        questions_data = await asyncio.to_thread(questions_file.read_text)
+        questions_cache = json.loads(questions_data)
+
+        # Extract only the question text for accuracy_relevance
+        if "accuracy_relevance" in questions_cache:
+            questions_cache["accuracy_relevance"] = questions_cache["accuracy_relevance"]["question_text"]
+
+    return questions_cache
+
+
+async def load_version_info() -> Dict[str, str]:
+    version_file_path = Path(__file__).parent.parent.parent / "version" / "version.json"
+    version_data = await asyncio.to_thread(version_file_path.read_text)
+    return json.loads(version_data)
+
+
+class UserFeedbackService:
+    """
+    The UserFeedbackService class provides the business logic for the user feedback routes
+    """
+
+    def __init__(self, db: AsyncIOMotorDatabase):
+        self.db = db
+
+    async def create_user_feedback(self, body: FeedbackRecordModel, user_info: UserInfo) -> FeedbackRecordModel:
+        try:
+            if not user_info or user_info.user_id != body.user_id:
+                raise HTTPException(status_code=403, detail="forbidden")
+
+            user_preference_repository = UserPreferenceRepository(self.db)
+            current_user_preferences = await user_preference_repository.get_user_preference_by_user_id(
+                user_info.user_id)
+            if current_user_preferences is None or body.session_id not in current_user_preferences.sessions:
+                raise HTTPException(status_code=403, detail="User does not have permission to access this session")
+
+            # Check if feedback for the given session_id already exists
+            existing_feedback = await self.db[Collections.USER_FEEDBACK].find_one({"session_id": body.session_id})
+            if existing_feedback:
+                raise HTTPException(status_code=409, detail=f"Feedback for session_id {body.session_id} already exists")
+
+            questions_data = await load_questions()
+            version_data = await load_version_info()
+            backend_version = f"{version_data['branch']}-{version_data['buildNumber']}"
+
+            # Validate the question IDs in the body
+            for item in body.feedback:
+                question_id = item.question_id
+                if question_id not in questions_data:
+                    raise HTTPException(status_code=400, detail=f"Invalid question ID: {question_id}")
+
```

Comment:
in case there is a selected_options array, we should also check that the items are part of the config for the specific questionid